docs/rules/vue-recommended/top-level-element-order.md

```diff
@@ -9,3 +9,97 @@ Checks if the component always uses `script`, `template`, and `style` tags' orde
 - **Readability:** A consistent order of elements improves readability. Developers can quickly locate specific sections of the component without having to search through the entire file.
 - **Maintainability:** When all components follow the same structure, it becomes easier to maintain and update the codebase. Developers can expect to find certain elements in specific locations, reducing cognitive load.
 - **Collaboration:** A standardized component structure facilitates better collaboration among team members. Everyone knows where to expect certain elements, leading to smoother code reviews and easier pair programming.
+
+## 😱 Examples of code for which this rule will throw a warning
+
+::: warning
+Single-File Components (SFCs) should maintain a consistent order of top-level elements: `<script>`, `<template>`, and `<style>`. The following examples illustrate cases where the order is incorrect or missing.
+:::
+
+### Example 1: Incorrect Order
+
+```javascript
+<template>
+  <h1 class="age-input">{{ age }}</h1>
+</template>
+
+<script setup>
+import { ref } from 'vue';
+let age = ref(22);
+</script>
+
+<style>
+.age-input {
+  color: white;
+}
+</style>
+```
+
+In this example, the `<template>` tag appears before the `<script>` tag, which violates the recommended order of top-level elements.
+
+### Example 2: Missing Top-Level Element
+```vue
+<template>
+  <h1 class="age-input">
+    22
+  </h1>
+</template>
+
+<style>
+.age-input {
+  color: white;
+}
+</style>
+```
+
+In this example, the `<script>` tag is missing entirely, which is also a violation of the required top-level element order.
+
+## 🤩 How to fix it?
+
+:::tip
+To comply with the recommended order, ensure that the top-level elements in your SFC are arranged as follows: `<script>`, followed by `<template>`, and then `<style>`.
+:::
+
+### Fixing Incorrect Order
+
+```vue
+<script setup>
+import { ref } from 'vue'
+const age = ref(22)
+</script>
+
+<template>
+  <h1 class="age-input">
+    {{ age }}
+  </h1>
+</template>
+
+<style>
```

Comment:
I would add scoped here

Reply:
fixed in https://github.com/rrd108/vue-mess-detector/pull/133/commits/c7841567f5b1592e0473650c19a6b5dc5ccc4a71